src/engine/SCons/Tool/lex.py

```diff
@@ -74,27 +75,26 @@ def get_lex_path(env, append_paths=False):
     """
     # save existing path to reset if we don't want to append any paths
     envPath = env['ENV']['PATH']
-    bins = ['lex', 'flex']
+    bins = ['win_flex', 'lex', 'flex']
```

Comment:
SHould win_flex only be searched for on win32?

Reply:
Currently this function is only ever called in win32 case, and the default paths also represent that, they all use paths with 'C:\'. In non win32 case, the system defaults from other platforms add some paths:  https://github.com/SCons/scons/blob/44740370071393c3de04a456506b9f2617350e05/src/engine/SCons/Platform/posix.py#L94

and then env.Detect method (as opposed to get_lex_path) locates flex or lex in this PR

Reply:
So the reason to search in that order is to not pick up the mingw or cygwin flex/lex?

Reply:
yeah on windows, you would prefer to use win_flex which is from chocolaty, over flex/lex from cygwin/mingw. win_flex will work the same for mingw environments as flex/lex, which the new live_mingw.py tests. I also tested independently mingw flex on windows environment, but that environment setup is not tested in the new tests I added. I suppose I could mock out the CHOCO_DEFAULT_PATH so that win_flex is not found and it would skip over to flex in an additional mingw test for our unit tests.

Reply:
I think you're missing what I'm saying. So if a user on windows has been using flex from mingw and also has win_flex installed, when they pull this update they'll get a different flex. So minimally we need to highlight this in the CHANGES.txt.  Though optimally we'd just add it to the end of the list.  If you're looking to make the tests work on the test infrastructure should be able to set LEX='win_flex' in the tests when on windows?

The key item is. We don't change things without telling the users in CHANGES.txt, and ideally we don't change things just so our ci environment works which could break in the users environment which worked with previous SCons.  Nothing more infuriating to a user than pulling and update, having your build break, no info in the changelog, and you have to go search sources to figure out what happened.. Then you send flame to the devs (rightfully so)..

Reply:
I didn't set the order to make any tests work. It works either way. I chose win_flex as the first default because if your on windows that seems like an appropriate default. I'll move it to the back to preserve existing defaults. 

In regards to the case you mentioned however, existing builds would be different with this PR even if win_flex is at the back of the list because they would be using the --no-unistd option in mingw builds now. So the existing default would require unistd.h on windows which is only possible in posix environment on windows like msys/cygwin, but would be broken in environments that don't support posix. This PR changes the default on windows to use --no-unistd option which works in either case, but will be different than before. Is that going to be an issue?

Reply:
do you need --no-unistd if using mingw's gcc? or only if using MSVC ?

Reply:
you don't need it with mingw, as it supports unistd.h, but it works with it as well. MSVC doesn't have unistd.h by default, so you need it.

Reply:
So from looking at the logic if your on win32 it's always going to have --nounistd? (which should really only be in place if not using cygwin or mingw?)

Reply:
I agree that would be a better default, but what would be a good way to know if the user intends to setup a posix environment. I thought about checking for these during lex generation, but then it would depend on the order you loaded tools?

Reply:
I'm guessing we'd be right the majority of the time if we checked for MS VC and only set -nounistd when that's being used.  We don't have clang-cl tool yet.   Maybe LEXNOUNISTD could be set in MSVC tool and if that's set then add it? That way tools can turn it on per tool, and users can request it as well?

Reply:
ok updated with MSVC tool setting no-unistd in 033661d